modules/openapi-generator/src/main/resources/dart/api_helper.mustache

```diff
@@ -50,3 +50,15 @@ String parameterToString(dynamic value) {
     return value.toString();
   }
 }
+
+/// Returns the decoded body by utf-8 if application/json with the given headers.
+/// Else, returns the decoded body by default algorithm of dart:http.
+/// Because avoid to text garbling when header only contains "application/json" without "; charset=utf-8".
+String _decodeBodyBytes(Response response) {
+  var contentType = response.headers['content-type'];
+  if (contentType != null && contentType.contains("application/json")) {
```

Comment:
@cimadai later we may want to further improve it to detect other JSON mime type, eg. https://github.com/OpenAPITools/openapi-generator/blob/1db105b6f9dc3058b6b8c17183d26719857e371a/samples/client/petstore/csharp-refactor/OpenAPIClient/src/Org.OpenAPITools/Client/ClientUtils.cs#L265 in C# to detect the JSON mime type using a regular expression